hudi-client/hudi-client-common/src/main/java/org/apache/hudi/client/BaseHoodieWriteClient.java

```diff
@@ -1031,6 +1031,7 @@ public void dropIndex(List<String> metadataPartitions) {
       if (metadataWriterOpt.isPresent()) {
         try (HoodieTableMetadataWriter metadataWriter = metadataWriterOpt.get()) {
           metadataWriter.dropMetadataPartitions(metadataPartitions);
+          metadataPartitions.forEach(partition -> table.getMetaClient().deleteIndexDefinition(partition));
```

Comment:
what happens if we fail after L1033.. and re try.. would it still work correctly/idempotent?

Reply:
Made a change to make sure the metadata changes are made before data deletion.